src/tcop/tcop.cpp

```diff
@@ -238,53 +241,92 @@ executor::ExecuteResult TrafficCop::ExecuteStatementPlan(
     }
 
     auto txn_result = txn->GetResult();
-    if (single_statement_txn == true || init_failure == true ||
+    if (single_statement_txn_ == true || init_failure == true ||
         txn_result == ResultType::FAILURE) {
-      auto &txn_manager = concurrency::TransactionManagerFactory::GetInstance();
-
       LOG_TRACE(
           "About to commit: single stmt: %d, init_failure: %d, txn_result: %s",
           single_statement_txn, init_failure,
           ResultTypeToString(txn_result).c_str());
       switch (txn_result) {
         case ResultType::SUCCESS:
-          // Commit
+          // Commit single statement
           LOG_TRACE("Commit Transaction");
-          p_status.m_result = txn_manager.CommitTransaction(txn);
+          p_status.m_result = CommitQueryHelper();
           break;
 
         case ResultType::FAILURE:
         default:
           // Abort
           LOG_TRACE("Abort Transaction");
-          p_status.m_result = txn_manager.AbortTransaction(txn);
-          curr_state.second = ResultType::ABORTED;
+          if (single_statement_txn_ == true) {
+            LOG_DEBUG("Tcop_txn_state size: %lu", tcop_txn_state_.size());
+            p_status.m_result = AbortQueryHelper();
+          } else {
+            tcop_txn_state_.top().second = ResultType::ABORTED;
+            p_status.m_result = ResultType::ABORTED;
+          }
       }
     }
   } else {
     // otherwise, we have already aborted
     p_status.m_result = ResultType::ABORTED;
   }
+  LOG_TRACE("Check Tcop_txn_state Size After ExecuteStatementPlan %lu", tcop_txn_state_.size());
   return p_status;
 }
 
 std::shared_ptr<Statement> TrafficCop::PrepareStatement(
     const std::string &statement_name, const std::string &query_string,
-    UNUSED_ATTRIBUTE std::string &error_message) {
+    UNUSED_ATTRIBUTE std::string &error_message,
+    const size_t thread_id UNUSED_ATTRIBUTE) {
   LOG_TRACE("Prepare Statement name: %s", statement_name.c_str());
   LOG_TRACE("Prepare Statement query: %s", query_string.c_str());
 
   std::shared_ptr<Statement> statement(
       new Statement(statement_name, query_string));
+  // transaction
+  // --multi-statements except BEGIN in a transaction
+  // ----not aborted
+  // ----COMMIT, ROLLBACK, and aborted
```

Comment:
Can you elaborate on these comments? They're helpful, but require more explanation.